doc/user/macros.rst

```diff
@@ -0,0 +1,189 @@
+Macros for kernels
+==================
+In addition to the CUDA/OpenCL portability macros in :file:`port.mako`,
+katsdpsigproc also provides a number of macros that help to write more advanced
+kernels.
+
+These macros make use of the more advanced parts of `Mako`_ syntax, and you
+should probably read its manual alongside this one.
+
+.. _Mako: https://docs.makotemplates.org
+
+Transposition
+-------------
+Transposition can be done with the :class:`.Transpose` operation without
+writing any kernel code; but one may wish to combine the transposition with
+other code to save memory bandwidth. The code to do this is contained in
+:file:`transpose_base.mako`, and can be loaded with
```

Comment:
Are these commands meant to link to the mako files? They are not clickable links

Reply:
I wasn't sure what to do with these references. I could possibly add links to the files in Github, but then the documentation isn't self-contained e.g. if someone makes their own fork or we migrate to Gitlab or something, the links won't automatically follow. Unfortunately sphinx.ext.viewcode seems very tied to Python code so I don't see an easy way to include the sources of these files into the built artefact.